scripts/generate-spec

```diff
@@ -44,7 +44,7 @@ sed -i \
 
 # Handle changelog
 if [ -e "${ORIG_SRC}/changelog" ]; then
-    sed -i '/@CHANGELOG@/ r changelog' "$1.tmp"
+    sed -i -e "/@CHANGELOG@/{r ${ORIG_SRC}/changelog" -e "d}" "$1.tmp"
```

Comment:
Now this will leave `@CHANGELOG@` if there is no `changelog` file..

Reply:
Sure but from my previous comment, what do we do when we have multiple spec? From your comment we can add "else" to simply change @CHANGELOG@ to %changelog. I would vote for the same changelog as it is in the same repo but there is also the problem with multiple different packages like in the installer.

Reply:
Actually, most packages in installer-qubes-os don't need generated spec and already have (outdated...) changelogs.
Anyway, yes, I'd put the same changelog for all packages. `.spec.in` should contain:
```
%changelog
@CHANGELOG@
```
Then, in case of no changelog (i.e. until generated changelog from git history got implemented) the `@CHANGELOG@` should be replaced by empty string, leaving empty `%changelog` section.

Reply:
Ok I do it and I correct all your suggestions on the builder to be prepare to open all the PR.

Reply:
Maybe I don't see something, but this still looks wrong. We have two cases:
- there is `changelog` file
- there is no `changelog` file

In the first case, `@CHANGELOG@` will be replaced with actual changelog (ok). In the second case, it will be replaced with `%changelog` keyword, resulting it duplicated `%changelog` in spec file.

Reply:
Hum no if I replace all %changelog in all spec.in?

Reply:
Yes, in that case it would work (as long as `changelog` file will contain `%changelog` keyword too). But IMO we should keep `%changelog` itself in spec.in. Generally I think the generator should fill _values only_, not keywords, headers etc. This way it would be much easier to follow spec structure.

Reply:
Ok. To summary: if not file, simply remove the @CHANGELOG@ and in all spec.in:

%changelog
@CHANGELOG@

Reply:
Yes.

Reply:
What about this one? That's the last one.

Reply:
Done. The list should be now empty...